src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsStartupLoader.cs

```diff
@@ -17,22 +23,61 @@ namespace Microsoft.AspNetCore.ApplicationInsights.HostingStartup
     /// </summary>
     public class ApplicationInsightsHostingStartup : IHostingStartup
     {
+        private const string ApplicationInsightsLoggerFactory = "Microsoft.ApplicationInsights.AspNetCore.Logging.ApplicationInsightsLoggerProvider";
+        private const string ApplicationInsightsSettingsFile = "ApplicationInsights.settings.json";
+
+        private static readonly KeyValuePair<string, string>[] _defaultLoggingLevels = new KeyValuePair<string, string>[]
+        {
+            new KeyValuePair<string, string>("Microsoft", "Warning"),
+            new KeyValuePair<string, string>("System", "Warning"),
+            new KeyValuePair<string, string>("Default", "Information")
+        };
+
         /// <summary>
         /// Calls UseApplicationInsights
         /// </summary>
         /// <param name="builder"></param>
         public void Configure(IWebHostBuilder builder)
         {
+            builder.ConfigureAppConfiguration((context, configurationBuilder) => ConfigureLogging(configurationBuilder));
             builder.UseApplicationInsights();
+
             builder.ConfigureServices(InitializeServices);
         }
 
+        private static void ConfigureLogging(IConfigurationBuilder configurationBuilder)
+        {
+            // Skip adding default rules when debugger is attached
+            // we want to sent all events to VS
+            if (!Debugger.IsAttached)
+            {
+                configurationBuilder.AddInMemoryCollection(GetDefaultLoggingSettings());
+            }
+
+            var home = Environment.GetEnvironmentVariable("HOME");
+            if (!string.IsNullOrEmpty(home))
+            {
+                var settingsFile = Path.Combine(home, "site", "diagnostics", ApplicationInsightsSettingsFile);
+                configurationBuilder.AddJsonFile(settingsFile, optional: true);
+            }
+        }
+
+        private static KeyValuePair<string, string>[] GetDefaultLoggingSettings()
+        {
+            return _defaultLoggingLevels.Select(pair =>
+                {
+                    var key = $"Logging:{ApplicationInsightsLoggerFactory}:LogLevel:{pair.Key}";
```

Comment:
Why did you extract just this part of the string as a constant when it's not used elsewhere?